### PR TITLE
Email settings accordion

### PR DESCRIPTION
**Pull request by ziggabyte**
## Description
This PR adds the accordion to the side of the editor, with settings for the editable components. 

## Screenshots
bild

## Changes
- `EmailSettings` component
- `BlockListItem` component, with sub-components for each block type
- function `checkBlockErrors` that takes a block and sees if there is errors in its data.

## Related issues
none

### CODE REVIEW

File: src/features/emails/components/EmailEditor/EmailSettings/utils/checkBlockErrors.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import isURL from 'validator/lib/isURL';
+import { OutputBlockData } from '@editorjs/editorjs';
+
+import { BLOCK_TYPES } from 'features/emails/types';
+
+export default function checkBlockErrors(block: OutputBlockData) {
```
Comment (richardolsson) [inline src/features/emails/components/EmailEditor/EmailSettings/utils/checkBlockErrors.ts:6]: Maybe this function could be called `blockHasErrors()` instead, to make it more clear what `true` and `false` means?

**Review by richardolsson — APPROVED** [review]
I have two minor suggestions for changes below, but if you don't agree with them I'm also ok with merging as is.

File: src/features/emails/l10n/messageIds.ts [hunk 2]
```diff
@@ -1,6 +1,33 @@
...
+        },
+      },
+      libraryImage: {
+        changeImage: m('Change image'),
+      },
+      titles: {
+        button: m('Button'),
+        libraryImage: m('Image'),
+        paragraph: m('Text'),
+      },
```
[the remark this answers is not in the document]
Comment (ziggabyte) [reply src/features/emails/l10n/messageIds.ts:28]: I agree, this i think is a leftover from something where i wanted to access the title strings via something like "....titles[tool.type]" but that is no longer happening. Will remove!

File: src/features/emails/components/EmailEditor/EmailSettings/utils/checkBlockErrors.ts (hunk 1 above)
Comment (ziggabyte) [reply src/features/emails/components/EmailEditor/EmailSettings/utils/checkBlockErrors.ts:6]: Great suggestion, thank you!

**Pull request merged**